File: backend/app/models/egg_offer.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Integer, DateTime
from ..database import Base


class EggOffer(Base):
    __tablename__ = "egg_offers"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    from_user_id = Column(String(36), nullable=False, index=True)
    to_user_id = Column(String(36), nullable=False, index=True)
    target_egg_id = Column(String(36), nullable=False)
    offered_egg_id = Column(String(36), nullable=False)
    offered_pinceaux = Column(Integer, default=0)
    status = Column(String(16), default="pending")  # pending, accepted, declined
    created_at = Column(DateTime, default=datetime.utcnow)
    responded_at = Column(DateTime, nullable=True)

File: mobile/lib/models/egg_offer.dart
class EggOffer {
  final String id;
  final String fromUserId;
  final String? fromUserPseudo;
  final String toUserId;
  final String? toUserPseudo;
  final String targetEggId;
  final String? targetEggDisplay;
  final String offeredEggId;
  final String? offeredEggDisplay;
  final int offeredPinceaux;
  final String status;
  final DateTime? createdAt;

  EggOffer({
    required this.id,
    required this.fromUserId,
    this.fromUserPseudo,
    required this.toUserId,
    this.toUserPseudo,
    required this.targetEggId,
    this.targetEggDisplay,
    required this.offeredEggId,
    this.offeredEggDisplay,
    this.offeredPinceaux = 0,
    this.status = 'pending',
    this.createdAt,
  });

  factory EggOffer.fromJson(Map<String, dynamic> json) {
    return EggOffer(
      id: json['id'] as String,
      fromUserId: json['from_user_id'] as String,
      fromUserPseudo: json['from_user_pseudo'] as String?,
      toUserId: json['to_user_id'] as String,
      toUserPseudo: json['to_user_pseudo'] as String?,
      targetEggId: json['target_egg_id'] as String,
      targetEggDisplay: json['target_egg_display'] as String?,
      offeredEggId: json['offered_egg_id'] as String,
      offeredEggDisplay: json['offered_egg_display'] as String?,
      offeredPinceaux: json['offered_pinceaux'] as int? ?? 0,
      status: json['status'] as String? ?? 'pending',
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'] as String)
          : null,
    );
  }
}
